ReactiveCocoa/Swift/Signal.swift

```diff
@@ -569,6 +569,88 @@ public func takeWhile<T, E>(predicate: T -> Bool)(signal: Signal<T, E>) -> Signa
 	}
 }
 
+/// Throttle values sent by the receiver for which `predicate` returns true,
+/// so that at least `interval` seconds pass between each, then forwards
+/// them on the given scheduler.
+/// 
+/// Values that do not pass `predicate`, Errors, and Completed events
+/// are forwarded immediately. If an unthrottled value is forwarded while
+/// a throttled value is pending, the throttled value is discarded.
+///
+/// If the receiver completes before the last throttled value is sent,
+/// it will be sent immediately.
+///
+/// If multiple values are received before the interval has elapsed, the
+/// latest value is the one that will be passed on.
+public func throttle<T, E>(interval: NSTimeInterval, onScheduler scheduler: DateSchedulerType, predicate: T -> Bool = { _ in true })(signal: Signal<T, E>) -> Signal<T, E> {
+	precondition(interval >= 0)
+
+	return Signal { observer in
+		let state = Atomic(ThrottleState<T>())
+		let flush = {
+			state.modify { state in
+				if let value = state.pendingValue {
+					let now = scheduler.currentDate
+					sendNext(observer, value)
```

Comment:
Forwarding events may be dangerous while holding the lock. Should try to hoist this outside of the modify block